python-checks/src/main/java/org/sonar/python/checks/hotspots/HardCodedCredentialsEntropyCheck.java

```diff
@@ -0,0 +1,214 @@
+
+  private static final String DEFAULT_RANDOMNESS_SENSIBILITY = "5.0";
+  @RuleProperty(
+    key = "randomnessSensibility",
+    description = "Allows to tune the Randomness Sensibility (from 0 to 10)",
+    defaultValue = DEFAULT_RANDOMNESS_SENSIBILITY)
+  public double randomnessSensibility = Double.parseDouble(DEFAULT_RANDOMNESS_SENSIBILITY);
+
+  private static final Pattern POSTVALIDATION_PATTERN = Pattern.compile("[a-zA-Z0-9_.+/~$-]([a-zA-Z0-9_.+/=~$-]|\\\\\\\\(?![ntr\"])){14,1022}[a-zA-Z0-9_.+/=~$-]");
+
+  private static final String MESSAGE = "\"%s\" detected here, make sure this is not a hard-coded secret.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, this::checkAssignment);
+    context.registerSyntaxNodeConsumer(Tree.Kind.ANNOTATED_ASSIGNMENT, this::checkAnnotatedAssignment);
+    context.registerSyntaxNodeConsumer(Tree.Kind.PARAMETER_LIST, this::checkParameterList);
+    context.registerSyntaxNodeConsumer(Tree.Kind.REGULAR_ARGUMENT, this::checkRegularArgument);
+    context.registerSyntaxNodeConsumer(Tree.Kind.DICTIONARY_LITERAL, this::checkDictionaryLiteral);
+  }
+
+  private void patternMatch(Name name, Tree location, String value, SubscriptionContext subscriptionContext) {
+    patternMatch(name.name(), location, value, subscriptionContext);
+  }
+
+  private void patternMatch(String name, Tree location, String value, SubscriptionContext subscriptionContext) {
+    if (!valuePassesPostValidation(value) || !entropyShouldRaise(value)) {
+      return;
+    }
+    patterns().stream()
+      .filter(pattern -> pattern.matcher(name).matches())
+      .findFirst()
+      .ifPresent(pattern -> subscriptionContext.addIssue(location, String.format(MESSAGE, name)));
+  }
+
+  private void checkParameterList(SubscriptionContext subscriptionContext) {
+    var parameterList = (ParameterList) subscriptionContext.syntaxNode();
+    parameterList.nonTuple().stream()
+      .filter(parameter -> parameter.name() != null)
+      .filter(parameter -> parameter.defaultValue() != null)
+      .filter(parameter -> parameter.defaultValue().is(Tree.Kind.STRING_LITERAL))
+      .forEach(parameter -> {
+        var value = ((StringLiteral) parameter.defaultValue()).trimmedQuotesValue();
+        patternMatch(parameter.name(), parameter.defaultValue(), value, subscriptionContext);
+      });
+  }
+
+  private void checkDictionaryLiteral(SubscriptionContext subscriptionContext) {
+    var dictionaryLiteral = (DictionaryLiteral) subscriptionContext.syntaxNode();
+    dictionaryLiteral.elements().stream().filter(e -> e.is(Tree.Kind.KEY_VALUE_PAIR))
+      .map(KeyValuePair.class::cast)
+      .filter(keyValuePair -> keyValuePair.value().is(Tree.Kind.STRING_LITERAL))
+      .forEach(keyValuePair -> {
+        var value = ((StringLiteral) keyValuePair.value()).trimmedQuotesValue();
+        var key = keyValuePair.key();
+        if (!key.is(Tree.Kind.STRING_LITERAL)) {
```

Comment:
this condition could be replaces with `.filter` stream method and you did it for keyValuePair.value() above